tests/integration/publish/publish_app_integ_base.py

```diff
@@ -0,0 +1,92 @@
+import os
+import json
+import uuid
+from unittest import TestCase
+import boto3
+
+try:
+    from pathlib import Path
+except ImportError:
+    from pathlib2 import Path
+
+
+class PublishAppIntegBase(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.region_name = "us-east-1"
+        cls.bucket_name = str(uuid.uuid4())
+        cls.bucket_name_placeholder = "<bucket-name>"
```

Comment:
Is this only to reset the files? Can we use templates and generate temp files? Someone writing a new test class will know to do this and I rather follow a consistent pattern that is always reproducible instead of re-writing to the same file each time.

Reply:
I ended up with generating a temp folder, writing to it based on the template files, then deleting the folder. I can't use tempfile because I need to pass the file path to the command, and [NamedTemporaryFile](https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile) doesn't work so well on Windows.

Reply:
You can use tempfile instead. Here is an example from our functional tests: https://github.com/awslabs/aws-sam-cli/blob/develop/tests/functional/function_code.py#L122